src/utils/s3_mods.py

```diff
@@ -0,0 +1,148 @@
+# General functions
+import json
+import pandas as pd
+from io import StringIO
+import logging
+from typing import List
+import subprocess
+import os
+
+# Functions specific to s3 bucket
+import boto3
+import raz_client
+from rdsa_utils.cdp.helpers.s3_utils import file_exists, create_folder_on_s3
+
+
+# set up logging
+s3_logger = logging.getLogger(__name__)
+
+
+def create_client(config):
+    client = boto3.client("s3")
+    raz_client.configure_ranger_raz(client, ssl_file=config["s3"]["ssl_file"])
+    return client
+
+
+# Read a CSV file into a Pandas dataframe
+def rd_read_csv(filepath: str, **kwargs) -> pd.DataFrame:
+    """Reads a csv from s3 bucket into a Pandas Dataframe using pydoop. 
+    If "thousands" argument is not specified, sets it to ",". 
+    Allows to use any additional keyword arguments of Pandas read_csv method.
+
+    Args:
+        filepath (str): Filepath (Specified in config)
+        kwargs: Optional dictionary of Pandas read_csv arguments
+    Returns:
+        pd.DataFrame: Dataframe created from csv
+    """
+    # Open the boto3 client
+    s3_client = config["client"]
+    with s3_client.get_object(
+        Bucket=config["s3"]["s3_bucket"],
+        Key=filepath
+    )['Body'] as file:
+        
+        # If "thousands" argument is not specified, set it to ","
+        if "thousands" not in kwargs:
+            kwargs["thousands"] = ","
+
+        # Read the scv file using the path and keyword arguments
+        try:
+            df = pd.read_csv(file, **kwargs)
+        except Exception:
+            if "usecols" in kwargs:
+                rd_logger.info("Columns not found: " + str(kwargs["usecols"]))
+            rd_logger.error(f"Could not read specified file: {filepath}")
```

Comment:
Please be more specific with this exception.
The logger error states that it could not read the specified file. This could not be the case.
For example, kwargs could be invalid.

Reply:
Added a logger message displaying all kwargs. Any more suggestions?

Reply:
Rather than use 'Exception', a more specific error type should be specified. pd.read_csv is capable of raising various different errors, therefore logging to the logger with such a specific message is inaccurate and will provide the user with false information.

A real world case here could be where the path passed to `pd.read_csv` is non existent, or contains data that causes `pd.read_csv` to error. However, the pipeline would simply catch the error (as it looks for any exception), and return one of the pre-defined error messages.